logfire/_internal/tracer.py

```diff
@@ -171,6 +171,12 @@ def record_exception(
         timestamp = timestamp or self.ns_timestamp_generator()
         record_exception(self.span, exception, attributes=attributes, timestamp=timestamp, escaped=escaped)
 
+    def __exit__(self, exc_type: type[BaseException] | None, exc_value: BaseException | None, traceback: Any) -> None:
+        if self.is_recording() and isinstance(exc_value, BaseException):
+            self.record_exception(exc_value, escaped=True)
```

Comment:
This is just the logic that used to be in `main._exit_span`. The parent class's version of this just calls `self.end()`, and I kept that below.